src/main/java/frc/robot/command/AutoAlignCommand.java

```diff
@@ -0,0 +1,217 @@
+      if (distance < minDistance) {
+        minDistance = distance;
+        closestMaster = master;
+      }
+    }
+
+    if (closestMaster == null) {
+      // Safety: if no master node is found, cancel command.
+      cancel();
+      return;
+    }
+
+    // Generate the absolute trajectory from the current pose to the closest master node.
+    absoluteTrajectory = PathPlanner.generateTrajectory(
+            currentPose,
+            List.of(), // no intermediate waypoints
+            closestMaster,
+            trajConfig
+    );
+
+    // Determine the desired sub node offset based on the selected mode.
+    Translation2d subOffset = subNodeOffsets.get(selectedMode);
+    // Compute final target for the second phase:
+    Pose2d subTarget = new Pose2d(closestMaster.getTranslation().plus(subOffset), closestMaster.getRotation());
+
+    // Generate a relative trajectory from the master node to the sub node.
+    // This trajectory is only translational; the rotation remains as the master node's rotation.
+    relativeTrajectory = PathPlanner.generateTrajectory(
+            closestMaster,
+            List.of(), // no intermediate waypoints
+            subTarget,
+            trajConfig
+    );
+
+    // Create trajectory following commands.
+    // Here we use PPSwerveControllerCommand (assumed to be configured similarly to your teleop drive controller)
+    Command absoluteFollow = new PPSwerveControllerCommand(
+            absoluteTrajectory,
+            positioning::getPose,
+            drive.kinematics,
+            drive.getDriveController(),
+            drive::driveSwerveStates,
+            drive
+    );
+
+    Command relativeFollow = new PPSwerveControllerCommand(
+            relativeTrajectory,
+            // For the relative trajectory, we use a lambda that returns the master node pose
+            // (the trajectory is relative; no additional rotation correction is needed)
+            () -> closestMaster,
+            drive.kinematics,
+            drive.getDriveController(),
+            drive::driveSwerveStates,
+            drive
+    );
+
+    // Wrap each trajectory-following command with a condition that checks if the driver has released the auto-align button.
+    Command absoluteFollowInterruptible = new ParallelRaceGroup(
+            absoluteFollow,
+            new WaitUntilCommand(() -> !isAutoAlignButtonHeld())
+                    .withTimeout(absoluteTrajectory.getTotalTimeSeconds() + 0.1)
+    );
+
+    Command relativeFollowInterruptible = new ParallelRaceGroup(
+            relativeFollow,
+            new WaitUntilCommand(() -> !isAutoAlignButtonHeld())
+                    .withTimeout(relativeTrajectory.getTotalTimeSeconds() + 0.1)
+    );
+
+    // Chain the two trajectory commands sequentially.
+    trajectoryCommand = new SequentialCommandGroup(absoluteFollowInterruptible, relativeFollowInterruptible);
+
+    // Schedule the internal trajectory command.
+    trajectoryCommand.schedule();
```

Comment:
We avoid scheduling commands from within other commands, it gets very messy.

This whole class could probably be replaced by [Commands#defer](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/Commands.html#defer(java.util.function.Supplier,java.util.Set))